### PR TITLE
feat: add unread badge to sidebar collapser

### PR DESCRIPTION
**Pull request by juliajforesti**
[SIDE-179](https://rocketchat.atlassian.net/browse/SIDE-179)

## Proposed changes (including videos or screenshots)

https://github.com/user-attachments/assets/77d265e5-8c74-4e14-a88b-cf7904831ee8

- [x] add unit tests
- [x] add e2e tests
## Further comments

[SIDE-179]: https://rocketchat.atlassian.net/browse/SIDE-179?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/sidebarv2/RoomList/RoomListCollapser.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	handleKeyDown: (e: React.KeyboardEvent<HTMLElement>, groupName: string) => void;
+	unreadCount: Pick<ISubscription, 'userMentions' | 'groupMentions' | 'unread' | 'tunread' | 'tunreadUser' | 'tunreadGroup'>;
+};
+const RoomListCollapser = ({
+	groupTitle,
+	unreadCount: unreadGroupCount,
+	collapsedGroups,
+	handleClick,
+	handleKeyDown,
+	...props
```
Comment (ggazzo) [inline apps/meteor/client/sidebarv2/RoomList/RoomListCollapser.tsx:21]: lets not use prop drilling, of {} ;)

File: apps/meteor/client/sidebarv2/RoomList/RoomList.tsx [hunk 2]
```diff
@@ -51,11 +52,12 @@ const RoomList = () => {
...
 				groupContent={(index) => (
-					<SidebarV2CollapseGroup
-						title={t(groupsList[index])}
-						onClick={() => handleClick(groupsList[index])}
-						onKeyDown={(e) => handleKeyDown(e, groupsList[index])}
-						expanded={!collapsedGroups.includes(groupsList[index])}
+					<RoomListCollapser
+						collapsedGroups={collapsedGroups}
+						handleClick={handleClick}
+						handleKeyDown={handleKeyDown}
```
Comment (aleksandernsilva) [inline apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: Could we keep the component event props as `onClick` and `onKeyDown`?
```suggestion
						onClick={handleClick}
						onKeyDown={handleKeyDown}
```

Comment (juliajforesti) [reply apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: The type is different from the default and I would have to omit from `HTMLAttributes`. Do you think it's really the best way? 
image

Comment (aleksandernsilva) [reply apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: I think omitting from `HTMLAttributes<HTMLElement>` makes sense regardless. If you leave as is the user can overwrite the event that calls `handleClick` and `handleKeyDown` which is not desired. 

I recommend doing the same you did with `onKeyDown` for `onClick` (e.g `onClick={(e) => onClick(e, groupTitle)}`) which will cover both the native and your custom event scenario.

**Review by ggazzo — APPROVED** [review]

**Pull request merged**